### PR TITLE
Typescript 4.1 ⚡️

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- Typescript 4.1
- Minor fixes for code wrinkles that 4.1 reveals

NOTE: Lint now exposes three warnings related to hook patterns that should get solved as follow-ups:

image

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [ ] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### CODE REVIEW

File: packages/components/src/Form/Inputs/InputText/InputText.tsx [hunk 1]
```diff
@@ -131,20 +144,9 @@ const InputTextLayout = forwardRef(
...
-      // need requestAnimationFrame here due to browser updating focus _after_ mousedown is called
-      window.requestAnimationFrame(() => {
-        internalRef.current && internalRef.current.focus()
-      })
-    }
-
     const mouseHandlers = {
       onClick,
-      onMouseDown: useWrapEvent(handleMouseDown, onMouseDown),
+      onMouseDown: onMouseDownWrapped,
```
Comment (lukelooker) [inline packages/components/src/Form/Inputs/InputText/InputText.tsx:149]: Rule of hooks got angry - conditionally loaded

File: packages/eslint-config/index.js [hunk 2]
```diff
@@ -52,18 +49,18 @@ module.exports = {
         'sort-keys': 'off',
       },
     },
+    {
+      files: ['*.ts', '*.tsx'],
+      rules: {
+        '@typescript-eslint/no-unused-vars': 'off',
```
Comment (lukelooker) [inline packages/eslint-config/index.js:55]: TSC itself covers our bases here. Turning this on makes React appear to be unloaded everywhere 🤷

File: packages/eslint-config/index.js [hunk 3]
```diff
@@ -52,18 +49,18 @@ module.exports = {
...
+    },
   ],
   parser: '@typescript-eslint/parser', // Specifies the ESLint parser
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true, // Allows for the parsing of JSX
-    },
-    ecmaVersion: 2018, // Allows for the parsing of modern ECMAScript features
-    sourceType: 'module', // Allows for the use of imports
-  },
```
Comment (lukelooker) [inline packages/eslint-config/index.js:63]: No longer needed (perhaps it never was needed?)

File: packages/eslint-config/index.js [hunk 4]
```diff
@@ -52,18 +49,18 @@ module.exports = {
...
   plugins: ['react-hooks', '@typescript-eslint', 'prettier', 'sort-keys-fix'],
   rules: {
     '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/no-use-before-define': 'off',
```
Comment (lukelooker) [inline packages/eslint-config/index.js:63]: We break this rule. 🤷

File: packages/eslint-config/index.js [hunk 5]
```diff
@@ -38,9 +38,6 @@ module.exports = {
     'plugin:@typescript-eslint/recommended',
     'prettier/@typescript-eslint',
     'plugin:prettier/recommended',
-    'plugin:import/errors',
-    'plugin:import/warnings',
-    'plugin:import/typescript',
```
Comment (lukelooker) [inline packages/eslint-config/index.js:43]: These were troublesome and turns out these bases are well covered by `standard` above

File: packages/components-theme-editor/src/ThemeEditorForm.tsx [hunk 6]
```diff
@@ -63,7 +63,7 @@ export const ThemeEditorForm: FC<ThemeEditorFormProps> = ({
   }
 
   return (
-    <Fieldset legend={name}>
+    <Fieldset>
```
[the remark this answers is not in the document]
Comment (dbchristopher) [reply packages/components-theme-editor/src/ThemeEditorForm.tsx:66]: 🤦

**Review by dbchristopher — APPROVED** [review]
It all works great for me! 💯

File: packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx [hunk 7]
```diff
@@ -157,9 +157,9 @@ const ComboboxListInternal = forwardRef(
     }, [optionsRef, isVisible, windowedOptions, windowedOptionsPropRef])
 
     const handleKeyDown = useKeyDown()
-    const handleBlur = isMulti
-      ? useBlur(ComboboxMultiContext)
-      : useBlur(ComboboxContext)
+    const useBlurSingle = useBlur(ComboboxContext)
+    const useBlurMulti = useBlur(ComboboxMultiContext)
+    const handleBlur = isMulti ? useBlurMulti : useBlurSingle
```
[the remark this answers is not in the document]
Comment (mdodgelooker) [reply packages/components/src/Form/Inputs/Combobox/ComboboxList.tsx:162]: I'd love to come up with a solution for this that doesn't involve calling `useBlur` 2x but I can't figure it out at the moment and it works just fine as is.

**Review by mdodgelooker — APPROVED** [review]
LGTM!

**Pull request merged**